src/main/java/org/terasology/signalling/componentSystem/SignalSwitchBehaviourSystem.java

```diff
@@ -195,19 +198,36 @@ public void handleDelayedTrigger(String actionId, EntityRef entity) {
                 });
     }
 
+    /**
+     * Updates the SignalSwitchBehaviorSystem.
+     * Handles pressure plate events and deletes old signal gate signal changes.
+     * @param delta The time in milliseconds since the last update
+     */
     @Override
     public void update(float delta) {
         handlePressurePlateEvents();
         deleteOldSignalChangesForGates();
     }
 
+    /**
+     * Stops a Producer from producing a signal if the event has the correct ActionId.
```

Comment:
This seems correct in spirit but perhaps with somewhat awkward wording. The method is an event handler (the `@ReceiveEvent`) and reacts to when a delayed trigger fires (essentially a scheduler) by stopping a signal if the action is releasing the switch, so to say. At least that's my interpretation, maybe @pollend can confirm :-)

So I would consider wording like "Event handler for when a delayed trigger fires with the intent to release a switch" (the "release a switch" is awkward as well and my interpretation - I figure it is akin to when you would release a physical button in-game or something like that - may be wrong!)

Reply:
I think this is for a button.